sdks/python/apache_beam/runners/portability/portable_runner.py

```diff
@@ -164,10 +165,19 @@ def add_runner_options(parser):
     all_options = self.options.get_all_options(
         add_extra_args_fn=add_runner_options,
         retain_unknown_options=self._retain_unknown_options)
+
     # TODO: Define URNs for options.
     # convert int values: https://issues.apache.org/jira/browse/BEAM-5509
+    def convert_pipeline_option_value(v):
+      if type(v) == int:
```

Comment:
why there is a special case for int? Can we do str(v) for everything except for valueproviders?

Reply:
Good question. I was just maintaining the status quo and adding a special case for `ValueProvider`. Let me see if I can dig up why the str(v) is there for ints

Reply:
Oh the relevant jira is linked right there. It seems that this is really a workaround for a bug in json_format where ints and floats aren't treated differently. I can look into fixing that upstream so we can get rid of the special case here.

I don't think it would be good to call str(v) on everything

Reply:
Interesting. For this PR, could you move that comment closer to the if type(v) == int line.

> I can look into fixing that upstream so we can get rid of the special case here.
If you have time, maybe file a bug. I do not think this is a very high priority for us.

> I don't think it would be good to call str(v) on everything
I agree. I was trying to understand why are we doing this.